Sources/StreamChat/Controllers/ChannelListController/ChannelListController.swift

```diff
@@ -42,12 +42,6 @@ public class ChatChannelListController: DataController, DelegateCallable, DataSt
             client.apiClient
         )
     
-    private var connectionObserver: EventObserver?
-    private let requestedChannelsLimit = 25
-
-    /// A Boolean value that returns wether pagination is finished
-    public private(set) var hasLoadedAllPreviousChannels: Bool = false
```

Comment:
Why did we remove this? We still need this right? How do we know the pagination has finished?

Reply:
Yep, good question 👍 

The use-case is: a user who loaded all the channels (100) goes offline and when connection comes back we refetch the first page (0..<20) of the channel query but user scrolls down the next page (20..<40) won't be loaded because of  `hasLoadedAllPreviousChannels == true`

That's not a final solution but a temp fix 🙂 I plan to reset the flag if we see `.remove` change reported by database observer but haven't yet addressed this

Reply:
I think we should try to fix this on this PR because then we will have a regression :/